store/kv/fdb.go

```diff
@@ -109,20 +109,23 @@ func (d *fdbkv) txWithRetry(ctx context.Context, fn func(fdb.Transaction) (inter
 	for {
 		tr, err := d.db.CreateTransaction()
 		if err != nil {
+			tr.Cancel()
 			return nil, err
 		}
-		defer tr.Cancel()
```

Comment:
What is the issue with defer?

Reply:
don't we need to cancel the `transaction` if we are not returning in the first attempt? `transaction` is getting created inside for loop but defer will only cancel the last one.

Reply:
Ah, right. This was a bug.

Reply:
Too many cancels doesn't look nice.
May be we can move the body of the loop the separate function and defer there.

Reply:
let me take a stab at it.